src/parsing/reverse-parsing-logic.ts

```diff
@@ -584,32 +644,80 @@ export function convertTrackerStructsToStrings(
         );
       }
     });
-
   const MappedTrackers = trackers
-    .filter(tracker => tracker.mapped)
+    .filter((tracker) => tracker.mapped)
     .map((tracker, iter) => {
-      const valueType = PT.find(pt => pt.enumeration === tracker.pType)?.name || "";
-      const keyType = PT.find(pt => pt.enumeration === tracker.trackerKeyType)?.name || "";
+      const valueType =
+        PT.find((pt) => pt.enumeration === tracker.pType)?.name || "";
+      const keyType =
+        PT.find((pt) => pt.enumeration === tracker.trackerKeyType)?.name || "";
+
+      var keys = [];
+
+      for (var key of mappedTrackerNames[iter].initialKeys) {
+        var decodedKey = "";
+        if (tracker.trackerKeyType == 0) {
+          decodedKey = decodeAbiParameters(
+            parseAbiParameters("address"),
+            key as `0x${string}`
+          )[0].toLowerCase();
+        } else if (tracker.trackerKeyType == 1) {
+          decodedKey = decodeAbiParameters(
+            parseAbiParameters("string"),
+            key as `0x${string}`
+          )[0];
+        } else if (tracker.trackerKeyType == 2) {
+          decodedKey = String(Number(key));
+        } else {
+          decodedKey = key;
+        }
+
+        keys.push(decodedKey);
+      }
+
+      var values = [];
+      for (var key of mappedTrackerNames[iter].initialValues) {
```

Comment:
are the routines inside the for loops starting on lines 679 and 657 the same? if so could thoughts on refactoring into a single functi0on and then using map to generate the array of values